src/components/pages/CompanyData/components/StatusInformation.tsx

```diff
@@ -40,60 +47,95 @@ export default function StatusInformation() {
   }
 
   const getStatusColor = (status: string | undefined) => {
-    if (status === SharingStateStatusType.Success) {
-      return 'success'
-    } else if (
-      status === SharingStateStatusType.Pending ||
-      status === SharingStateStatusType.Initial
-    ) {
-      return 'warning'
-    } else {
-      return 'error'
-    }
+    if (status === SharingStateStatusType.Success) return 'success'
+    else if (status === SharingStateStatusType.Initial) return 'warning'
+    else if (status === SharingStateStatusType.Pending) return 'info'
+    else if (status === SharingStateStatusType.Ready) return 'primary'
+    return 'error'
```

Comment:
In discussion with Lavanya we came up with a solution using a map instead of a function, which is more elegant imho. Can you please have a look and implement accordingly:
https://github.com/eclipse-tractusx/portal-frontend/pull/967/files#diff-ef36c5be86b0e58f10f5281ace1652ca5c2a443d542d2336fd3b0fd28f945de7R101

Reply:
I have already used this type of mapping just above this method. 

```
const statusColor: Record<string, 'default' | 'primary' | 'secondary' | 'error' | 'info' | 'success' | 'warning'> = {
    Success: 'success',
    Pending: 'info',
    Ready: 'primary',
    Initial: 'warning',
    Error: 'error'
  }
 ```
  
 Could you pls verify the above one?

Reply:
I haven't seen that in your code - now I updated your PR with a shared map to get the colors from status types using proper types instead of strings